cvat/apps/lambda_manager/views.py

```diff
@@ -0,0 +1,307 @@
+            # TODO: handle exceptions
+            if serializer.is_valid(raise_exception=True):
+                patch_task_data(db_task.id, results, "create")
+
+
+def return_response(success_code=status.HTTP_200_OK):
+    def wrap_response(func):
+        @wraps(func)
+        def func_wrapper(*args, **kwargs):
+            data = None
+            status_code = success_code
+            try:
+                data = func(*args, **kwargs)
+            except requests.ConnectionError as err:
+                status_code = status.HTTP_503_SERVICE_UNAVAILABLE
+                data = str(err)
+            except requests.HTTPError as err:
+                status_code = err.response.status_code
+                data = str(err)
+            except requests.Timeout as err:
+                status_code = status.HTTP_504_GATEWAY_TIMEOUT
+                data = str(err)
+            except requests.RequestException as err:
+                status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+                data = str(err)
+            except ValidationError as err:
+                status_code = err.code
+                data = err.message
+
+            return Response(data=data, status=status_code)
+
+        return func_wrapper
+    return wrap_response
+
+class FunctionViewSet(viewsets.ViewSet):
+    lookup_value_regex = '[a-zA-Z0-9_.-]+'
+    lookup_field = 'id'
+
+    @return_response()
+    def list(self, request):
+        gateway = LambdaGateway()
+        return [f.to_dict() for f in gateway.list()]
+
+    @return_response()
+    def retrieve(self, request, id):
+        gateway = LambdaGateway()
+        return gateway.get(id).to_dict()
+
+    @return_response()
+    def call(self, request, id):
+        try:
+            task = request.data['task']
+            points = request.data.get('points')
+            frame = request.data['frame']
+            db_task = TaskModel.objects.get(pk=task)
+        except (KeyError, ObjectDoesNotExist) as err:
+            raise ValidationError(
+                '`{}` lambda function was run '.format(id) +
+                'with wrong arguments ({})'.format(str(err)),
+                code=status.HTTP_400_BAD_REQUEST)
+
+        gateway = LambdaGateway()
+        lambda_func = gateway.get(id)
+
+        return lambda_func.invoke(db_task, frame, points)
+
+class RequestViewSet(viewsets.ViewSet):
+    @return_response()
+    def list(self, request):
+        queue = LambdaQueue()
+        return [job.to_dict() for job in queue.get_jobs()]
+
+    @return_response()
+    def create(self, request):
+        function = request.data['function']
+        threshold = request.data.get('threshold')
+        task = request.data['task']
+
+        gateway = LambdaGateway()
+        queue = LambdaQueue()
+        lambda_func = gateway.get(function)
+        job = queue.enqueue(lambda_func, threshold, task)
+
+        return job.to_dict()
+
+    @return_response()
+    def retrieve(self, request, pk):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)